pkg/provider/kubernetes/gateway/kubernetes.go

```diff
@@ -706,7 +706,7 @@ func getAllowedRouteKinds(gateway *gatev1.Gateway, listener gatev1.Listener, sup
 	}
 
 	var (
-		routeKinds = []gatev1.RouteGroupKind{}
```

Comment:
If I remember correctly these assignments were there for a reason. 🤔

Reply:
conformance tests are green, don't think the change had an impact.

Reply:
Yeah, I just saw it. I'm trying hard to remember why it was this way. There was a reason. Maybe I ran a test that is still skipped in CI.

Reply:
I think it was this one. It is currently skipped in CI: https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/tests/gateway-invalid-route-kind.go#L46

Reply:
You are totally right @pascal-hofmann!!

Very sorry about this change.

My last commit fix that